OpenRA.Mods.Common/Traits/Air/AttackPlane.cs

```diff
@@ -40,7 +40,9 @@ public override Activity GetAttackActivity(Actor self, Target newTarget, bool al
 		protected override bool CanAttack(Actor self, Target target)
 		{
 			// Don't fire while landed or when outside the map.
-			return base.CanAttack(self, target) && self.CenterPosition.Z > 0 && self.World.Map.Contains(self.Location);
+			return base.CanAttack(self, target)
+				&& self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0
```

Comment:
`&& self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length >= aircraft.MinAirborneAltitude` where `readonly AircraftInfo aircraft` and `aircraft = self.Info.TraitInfo<AircraftInfo>()` in the constructor.